src/components/PresentBar/PresentBar.tsx

```diff
@@ -14,26 +14,69 @@
  * limitations under the License.
  */
 
+import ArrowBackIosNewIcon from '@mui/icons-material/ArrowBackIosNew';
+import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
 import PresentToAllIcon from '@mui/icons-material/PresentToAll';
 import { Box } from '@mui/material';
+import { useCallback } from 'react';
 import { useTranslation } from 'react-i18next';
-import { PresentationState, usePresentationMode } from '../../state';
+import {
+  PresentationState,
+  useActiveSlide,
+  useActiveWhiteboardInstance,
+  useActiveWhiteboardInstanceSlideIds,
+  usePresentationMode,
+} from '../../state';
 import { useUserDetails } from '../../store';
-import { Toolbar, ToolbarAvatar, ToolbarToggle } from '../common/Toolbar';
+import {
+  Toolbar,
+  ToolbarAvatar,
+  ToolbarButton,
+  ToolbarToggle,
+} from '../common/Toolbar';
 
 export function PresentBar() {
   const { t } = useTranslation();
-
+  const slideIds = useActiveWhiteboardInstanceSlideIds();
+  const whiteboardInstance = useActiveWhiteboardInstance();
+  const { activeSlideId, isFirstSlideActive, isLastSlideActive } =
+    useActiveSlide();
   const { state, togglePresentation } = usePresentationMode();
 
+  const isViewingPresentation = state.type === 'presenting';
+
+  const handleToNextSlideClick = useCallback(() => {
+    if (activeSlideId) {
+      const activeSlideIndex = slideIds.indexOf(activeSlideId);
```

Comment:
You can also get the slideIds from the `whiteboardInstance`. Then we don't need another hook call that potentially triggers a rendering:

```suggestion
      const slideIds = whiteboardInstance.getSlideIds();
      const activeSlideIndex = slideIds.indexOf(activeSlideId);
```